sql/database_test.go

```diff
@@ -135,7 +135,7 @@ func TestDatabaseVacuumState(t *testing.T) {
 
 	db, err = Open("file:"+dbFile,
 		WithMigrations([]Migration{}),
-		WithVacuumState(9),
+		WithVacuumState(1000),
```

Comment:
The problem with the test is that the first `Open` applies the actual migrations that are defined for `go-spacemesh` while the second doesn't apply any migrations at all. I feel like a more reasonable version of the test would look like this:

```go
func TestDatabaseVacuumState(t *testing.T) {
	dir := t.TempDir()

	ctrl := gomock.NewController(t)
	migration1 := NewMockMigration(ctrl)
	migration1.EXPECT().Order().Return(1).AnyTimes()
	migration1.EXPECT().Apply(gomock.Any()).Return(nil).Times(1)

	migration2 := NewMockMigration(ctrl)
	migration2.EXPECT().Order().Return(2).AnyTimes()
	migration2.EXPECT().Apply(gomock.Any()).Return(nil).Times(1)

	dbFile := filepath.Join(dir, "test.sql")
	db, err := Open("file:"+dbFile,
		WithMigrations([]Migration{migration1}),
	)
	require.NoError(t, err)
	require.NoError(t, db.Close())

	db, err = Open("file:"+dbFile,
		WithMigrations([]Migration{migration1, migration2}),
		WithVacuumState(2),
	)
	require.NoError(t, err)
	require.NoError(t, db.Close())

	// we run pragma wal_checkpoint(TRUNCATE) after vacuum, which drops the wal file
	_, err = os.Open(dbFile + "-wal")
	require.ErrorIs(t, err, os.ErrNotExist)
}
```

This doesn't use any of the real migrations while still verifying that `WithVacuumState` works correctly.

Reply:
Sure - can do 👍

Reply:
PR with updated tests: https://github.com/spacemeshos/go-spacemesh/pull/5508